pkg/network/node/vnids.go

```diff
@@ -204,6 +219,11 @@ func (vmap *nodeVNIDMap) handleAddOrUpdateNetNamespace(obj, _ interface{}, event
 	netns := obj.(*networkapi.NetNamespace)
 	klog.V(5).Infof("Watch %s event for NetNamespace %q", eventType, netns.Name)
 
+	// Skip this event if NetID already exists under different netns name
+	if vmap.findDuplicateNetID(netns.NetName, netns.NetID) {
```

Comment:
you should update `handleDeleteNetNamespace` too, to not call `vmap.policy.DeleteNetNamespace` if `vmap.unsetVNID` failed (since `policy.DeleteNetNamespace` might do the wrong thing if you pass it a NetNamespace with a duplicate vnid)